src/movies.js

```diff
@@ -1,16 +1,79 @@
-// Iteration 1: All directors? - Get the array of all directors.
-// _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
+
+
+// Iteration 1: All directors? - Get the moviesay of all directors.
+
+function getAllDirectors(movies){ 
+  const allDirectors = movies.map(function(mov){
+    return mov.director;
+  })
+  return allDirectors;
+}
+
+// _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the moviesay of directors. How could you "clean" a bit this moviesay and make it unified (without duplicates)?
+
+function findUniqueDirectors(movies) {
+  const uniqueDirectors = [...new Set(getAllDirectors(movies))];
+  return uniqueDirectors ;
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies (movies){ 
+  const numberOfDramas = movies.filter(function(movies){
+    return movies.director.includes('Steven Spielberg') && movies.genre.includes('Drama');
+  })
+  return numberOfDramas.length;  
+}
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
-// Iteration 4: Drama movies - Get the average of Drama Movies
+function ratesAverage(movies){
+   
+  const sumRating = movies.reduce(function(acc, val){
+    if (movies.length === 0) return 0;
```

Comment:
I would suggest to move this part outside the reduce method - to the beginning of the function.